src/pyJiraCli/cmd_profile.py

```diff
@@ -0,0 +1,221 @@
+                            help="The server url for the profile.")
+
+
+    login_group.add_argument("--token",
+                             type=str,
+                             metavar="<api token>",
+                             help="The api token for login with this server profile")
+
+    login_group.add_argument('--cert',
+                            type=str,
+                            metavar="<certificate path>",
+                            required=False,
+                            help="The server url for the profile.")
+
+    datatype_desc = sub_parser_profile.add_argument_group(
+        title='profile operations',
+        description='Only one operation type can be processed at a time.'
+    )
+
+    option_grp = datatype_desc.add_mutually_exclusive_group(required=True)
+
+    option_grp.add_argument('--add',
+                           '-a',
+                            action="store_true",
+                            help="Add a new server profile.")
+
+    option_grp.add_argument('--remove',
+                           '-r',
+                            action="store_true",
+                            help="Delete an existing server profile.")
+
+    option_grp.add_argument('--update',
+                           '-u',
+                            action="store_true",
+                            help="Update an existing server profile with new data.")
+
+    return sub_parser_profile
+
+def execute(args) -> Ret.CODE:
+    """ This function servers as entry point for the command 'profile'.
+        It will be stored as callback for this moduls subparser command.
+    
+    Args: 
+        args (obj): The command line arguments.
+        
+    Returns:
+        Ret.CODE:   Returns Ret.RET_OK if successful or else the corresponding error code.
+    """
+    return _cmd_profile(args)
+
+def _cmd_profile(args) -> Ret.CODE:
+    """ Process the 'profile' command and its 
+        commandline arguments.
+
+    Args:
+        args (obj): The commandline arguments.
+
+    Returns:
+        Ret.CODE: The return status of the module.
+    """
+
+    ret_status = Ret.CODE.RET_OK
+
+    if args.add:
+        ret_status = _add_profile(args)
+
+    elif args.remove:
+        ret_status = _remove_profile(args.profile_name)
+
+    else:
+        ret_status = _update_profile(args)
+
+    return ret_status
+
+def _add_profile(args) -> Ret.CODE:
+    """ Adds a new profile to the configuration using provided arguments.
+
+    Args:
+        args (obj): Object containing the commandline arguments for profile addition.
+
+    Returns:
+        Ret.CODE: Status code indicating the success or failure of the profile addition.
+    """
+    ret_status = Ret.CODE.RET_OK
+
+    _server = Server()
+    _profile = Profile()
+
+    if args.url is None:
+        ret_status = Ret.CODE.RET_ERROR_NO_SERVER_URL
+
+    else:
+        name = args.profile_name
+        url = args.url
+        token = args.token
```

Comment:
Its is not checked whether a token or a cert is given. Only the url is being checked

Reply:
This is because only the url and the profilename is requiered, the token and certificate are optional and can also be added later with the --update option. In verbose mode a warning will be displayed if token and certificate are not given.